src/retry-queue.js

```diff
@@ -0,0 +1,108 @@
+import { RequestQueueScaffold } from './base-request-queue'
+import { encodePostData, xhr } from './send-request'
+
+export class RetryQueue extends RequestQueueScaffold {
+    constructor(captureMetrics) {
+        super()
+        this.captureMetrics = captureMetrics
+        this.isPolling = false
+        this._requestRetriesMap = {} // <RequestId, number>
+        this._counterToQueueMap = {}
```

Comment:
I don't really understand what is the `counterToQueueMap` just by reading the variable. Could there be a more descriptive name for this, even if more verbose?

Some ideas for clarification, though comments below might invalidate these thoughts:

```
this._requestRetriesMap --> this._retryCountForRequest
this._counterToQueueMap --> this._requestsToPerformAtRunStep
```